### ISSUE TITLE
OSM editing plugin: support moving POI

### ISSUE THREAD

**Issue by christophgysin** [osmandapp/Osmand#3459]
I use Osmand to improve OSM while travelling. While it is easy to add and modify existing POIs, it seems currently not possible to move a POI that is in a wrong location. Currently, the only way is to delete the POI and create a new one at the correct location (with all the previous tags). While this is cumbersome, it also makes it hard to follow the history of the POI.

I suggest to support moving POIs similarly to how Favorites can be moved.

**Comment by vshcherb** [comment]
It is possible to move only pois which are not uploaded.

**Comment by christophgysin** [comment]
As reported in #3460, moving newly created POIs doesn't work either.

**Comment by KOLANICH** [comment]
>It is possible to move only pois which are not uploaded.

It is also possible to move POIs which are uploaded. (I use osmand from fdroid, not from play store)

**Comment by stephan75** [comment]
By the way:
What about the situation when editing a POI in Osmand and uploading it, when that POI has been deleted in main OSM database in the meantime?

**Comment by christophgysin** [comment]
@KOLANICH I just tried with the latest nightly build, and I still can't move POIs. What version are you using? How do you manage to move POIs?

**Comment by KOLANICH** [comment]
1 tap on poi
2 tap and hold the orange marker (that one which appears if you tap and hold on empty place on a map)
3 move the map relative to the marker

**Comment by christophgysin** [comment]
Yes, that's exactly how I tried.

I just installed the version from F-droid, and I'm still unable to move existing POIs. Are you sure we are talking about the same? Moving existing POIs from a downloaded map?

**Comment by KOLANICH** [comment]
I have retried, it really doesn't work. But I found why have it worked for me: I usually not just move a POI, but also add some missing info. So the workaround is to edit the poi and then move the edited one and then delete the edits if they are not needed.

**Comment by Lensi** [comment]
@KOLANICH That workaround seems to work on the surface, but the actual coordinate changes are not saved.

**Comment by bmw216** [comment]
Implement this feature!!! 2026

**Comment by matthijskooijman** [comment]
~~I've noticed that there is a "Change marker position" button for POIs, but it is greyed out. However, when you select "Modify POI" and then "Save" (even without making any changes), then selecting the modified POI allows selecting the "Change marker position" button and modify the position. Sounds bugged, but this is a workaround that works for me.~~ Turns out this does not actually work, see below.

**Comment by mnalis** [comment]
> I've noticed that there is a "Change marker position" button for POIs, but it is greyed out

It is not just for moving POIs; it allows you to move your Markers[^1] (flags), Favourites, Photos, Audio notes, parking location, and ***unuploaded*** (i.e. newly created) POIs and OSM notes. 

> However, when you select "Modify POI" and then "Save" (even without making any changes), then selecting the modified POI allows selecting the "Change marker position" button and modify the position

Are you **sure** that it _actually works_? See the above _"That workaround seems to work on the surface, but the actual coordinate changes are **not** saved"_.  I've tried [here](https://www.openstreetmap.org/node/11179163931/history/4) with OsmAnd 5.3.10 to make an "empty" change and then move the POI, and it didn't _actually_ change the coordinates (even if it claimed that it did).

It would be nice if that actually worked that way (as just allowing to move any random POI just by misclicking without first making an edit might cause a lot of unintentional damage on the map), but it makes sense why it might trigger the bug -- once you've edited it, you now have ***unuploaded*** (i.e. modified) POI, so it behave like any other unuploaded modified POI (i.e. one you've just created, and on which changing coordinates is allowed).

[^1]: thus the name _"Change **marker** position"_, I guess

**Comment by matthijskooijman** [comment]
> Are you sure that it actually works? See the above "That workaround seems to work on the surface, but the actual coordinate changes are not saved". I've tried [here](https://www.openstreetmap.org/node/11179163931/history/4) with OsmAnd 5.3.10 to make an "empty" change and then move the POI, and it didn't actually change the coordinates (even if it claimed that it did).

Oh bloody hell, you're right. Just tested this and produced this changeset, which changes absolutely nothing... https://www.openstreetmap.org/changeset/188576968

Seems I've done dozens of changes over the past years assuming they were applied, but turns out I was just generating pointless noise. Never actually checked the results, I usually did this with POIs that I was passing by (and the marker on the map was not updated, but I assumed it was because it needed to go via the (live) map updates first...).

Thanks for pointing this out.

So, I second the request for implementing this, then :-)
